GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -401,76 +394,58 @@ public async Task<DataTable> GetConfiguratorDataAsync(string name)
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS maximumValue ON maximumValue.item_id = mainStep.id AND maximumValue.`key` = 'max_value'
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS validationRegex ON validationRegex.item_id = mainStep.id AND validationRegex.`key` = 'validation_regex'
-
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS requiredErrorMessage ON requiredErrorMessage.item_id = mainStep.id AND requiredErrorMessage.`key` = 'required_error_message'
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS minimumValueErrorMessage ON minimumValueErrorMessage.item_id = mainStep.id AND minimumValueErrorMessage.`key` = 'min_value_error_message'
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS maximumValueErrorMessage ON maximumValueErrorMessage.item_id = mainStep.id AND maximumValueErrorMessage.`key` = 'max_value_error_message'
-    LEFT JOIN {WiserTableNames.WiserItemDetail} AS validationRegexErrorMessage ON validationRegexErrorMessage.item_id = mainStep.id AND validationRegexErrorMessage.`key` = 'validation_regex_error_message'
+            var configuratorSettings = await databaseConnection.GetAsync($@"SELECT
+    configurator.id AS configuratorId,
+    CONCAT_WS('', mainTemplate.`value`, mainTemplate.long_value) AS mainTemplate,
+    CONCAT_WS('', progressBarTemplate.`value`, progressBarTemplate.long_value) AS progressBarTemplate,
+    CONCAT_WS('', progressBarStepTemplate.`value`, progressBarStepTemplate.long_value) AS progressBarStepTemplate,
+    CONCAT_WS('', summaryTemplate.`value`, summaryTemplate.long_value) AS summaryTemplate,
+    priceCalculationQuery.`value` AS priceCalculationQuery,
+    deliveryTimeCalculationQuery.`value` AS deliveryTimeCalculationQuery
+FROM {WiserTableNames.WiserItem} AS configurator
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS mainTemplate ON mainTemplate.item_id = configurator.id AND mainTemplate.`key` = 'template'
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS progressBarTemplate ON progressBarTemplate.item_id = configurator.id AND progressBarTemplate.`key` = 'progress_bar_template'
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS progressBarStepTemplate ON progressBarStepTemplate.item_id = configurator.id AND progressBarStepTemplate.`key` = 'progress_bar_step_template'
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS summaryTemplate ON summaryTemplate.item_id = configurator.id AND summaryTemplate.`key` = 'summary_template'
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS priceCalculationQuery ON priceCalculationQuery.item_id = configurator.id AND priceCalculationQuery.`key` = 'price_calculation_query'
+LEFT JOIN {WiserTableNames.WiserItemDetail} AS deliveryTimeCalculationQuery ON deliveryTimeCalculationQuery.item_id = configurator.id AND deliveryTimeCalculationQuery.`key` = 'delivery_time_calculation_query'
+WHERE configurator.moduleid = {Constants.ConfiguratorModuleId} AND configurator.entity_type = '{Constants.ConfiguratorEntityType}' AND configurator.title = ?name");
```

Comment:
You shouldn't use `moduleid` like this. Module IDs are dynamic, they can be different for each project. In one project the configurator module could have ID 5 and in another project it could be 6. Why do you even check for moduleid at all here? We never do that in queries, there's usually no point because an entity type usually only exists in only one module. Also, there is no index on `wiser_item` that uses both `moduleid` and `entity_type`.